src/frontends/pytorch/src/op/flatten.cpp

```diff
@@ -19,11 +20,13 @@ namespace op {
 using namespace ov::op;
 
 OutputVector translate_flatten(const NodeContext& context) {
-    num_inputs_check(context, 1, 3);
-    auto x = context.get_input(0);
+    num_inputs_check(context, 1, 3, true);  // allow_complex = true
+    auto [x, complex] = unwrap_complex(context.get_input(0));
+
     Output<Node> shape;
     Output<Node> rank;
-    std::tie(shape, rank) = get_shape_rank(context, x, true);
+    // Use original input for complex-aware rank calculation
+    std::tie(shape, rank) = get_shape_rank(context, context.get_input(0), true);
```

Comment:
```suggestion
    std::tie(shape, rank) = get_shape_rank(context, x, true);
```
You need get shape of `x` after `ComplexTypeMark` was removed